runtime/memory/generic.c

```diff
@@ -22,6 +22,19 @@ void expand_memory() {
     memory_size += WASM_PAGE_SIZE;
 }
 
+i32 instruction_memory_size() {
+    return memory_size / WASM_PAGE_SIZE;
+}
+
+i32 instruction_memory_grow(i32 count) {
+    i32 prev_size = instruction_memory_size();
+    for (int i = 0; i < count; i++) {
+        expand_memory();
+    }
+
+    return prev_size;
+}
+
```

Comment:
In retrospect, I really should've abstracted this repeated code out... this new addition is in line with everything else tho, so keep it this way